src/main/java/org/ibp/api/rest/inventory_new/TransactionResource.java

```diff
@@ -99,4 +107,19 @@ public long getCount() {
 
 	}
 
+	@ApiOperation(value = "Create Transaction", notes = "Create a new transaction")
+	@RequestMapping(value = "/crops/{cropName}/lots/{lotId}/transactions", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<Integer> createTransaction(
+		@PathVariable final String cropName,
+		@PathVariable final String lotId,
+		@ApiParam("Transaction to be created")
+		@RequestBody final TransactionDto transactionDto) {
+
+
+		final String userName = SecurityUtil.getLoggedInUserName();
+		final WorkbenchUser user = this.userService.getUserByUsername(userName);
+		transactionDto.setUser(user.getUserid().toString());
+		return new ResponseEntity<>(this.transactionService.saveTransaction(transactionDto, lotId, TransactionStatus.COMMITTED), HttpStatus.CREATED);
```

Comment:
@vmaletta TransactionDto already contains a lot so, you could set the lotId to TransactionDTO and avoid passing the lotId param. Also TransactionStatus should not be passed as a param. It should be inferred. All Deposits have Committed status